test/TestMetadataParserLib.sol

```diff
@@ -99,6 +99,7 @@ contract JBDelegateMetadataLib_Test is Test {
             _datas[_i] = abi.encode(type(uint256).max - _i);
         }
 
+        if (_numberOfIds == 220) vm.expectRevert(abi.encodeWithSignature("METADATA_TOO_LONG()"));
```

Comment:
where does the number 220 come from here?

Reply:
@mejango 
offsets are counted from the beginning of the whole metadata: accounting for a lookup table which is now fitting these 219 entries -> len(id, offset) = 5B,  times 219 entries, its 35 words (ceiling div), then another word is "lost" for the initial reserved, we spend 36 words for the initial part, leaving 219 possible offsets

Reply:
Because of this, I couldn't find a way to (re)produce the bug underlying this pr (even with 2 really big data/offset for instance), but maybe there is a solution to this system (cost of fix being minor, I wouldn't spend time trying to do so anyway - fix and move on imo)

Reply:
ok. might could use a comment in the test explaining the 220. 
so you approve of this PR as is @drgorillamd ?